test/openAPI/features/support/wallet_generate_link_code.js

```diff
@@ -140,19 +154,150 @@ When(
+  () =>
+    specWalletGenerateLinkCodeReused
+      .post(baseUrl)
+      .withHeaders(X_XSRF_TOKEN.key, X_XSRF_TOKEN.value)
+      .withCookies(X_XSRF_TOKEN.key, X_XSRF_TOKEN.value)
+      .withJson({
+        requestTime: new Date().toISOString(),
+        request: {
+          transactionId: 'transaction_id_003',
+        },
+      })
+);
+
+Then(
+  'Receive a response for completed transactionId from the \\/linked-authorization\\/link-code endpoint',
+  async () => await specWalletGenerateLinkCodeReused.toss()
+);
+
+Then(
+  'The \\/linked-authorization\\/link-code endpoint response for completed transactionId should have status 200',
+  () => specWalletGenerateLinkCodeReused.response().to.have.status(200)
+);
+
+Then(
+  'The \\/linked-authorization\\/link-code endpoint response for completed transactionId should have content-type: application\\/json header',
+  () =>
+    specWalletGenerateLinkCodeReused
+      .response()
+      .should.have.header(contentTypeHeader.key, contentTypeHeader.value)
+);
+
+Then(
+  'The \\/linked-authorization\\/link-code endpoint response for completed transactionId should match json schema with errors',
+  () => {
+    chai
+      .expect(specWalletGenerateLinkCodeReused._response.json)
+      .to.be.jsonSchema(walletGenerateLinkCodeResponseSchema);
+    chai.expect(specWalletGenerateLinkCodeReused._response.json.errors).to.not
+      .be.empty;
+  }
+);
+
+Then(
+  'The \\/linked-authorization\\/link-code response for completed transactionId should contain errorCode property equals to {string}',
+  (errorCode) =>
+    chai
+      .expect(
+        specWalletGenerateLinkCodeReused._response.json.errors
+          .map((error) => error.errorCode)
+          .toString()
+      )
+      .to.be.equal(errorCode)
+);
+
+// Scenario: Not able to generate link code because of invalid requestTime
+// Given and others Then for this scenario are written in the aforementioned example
+When(
+  'Send POST \\/linked-authorization\\/link-code request with given X-XSRF-TOKEN header, transactionId and invalid requestTime',
+  () =>
     specWalletGenerateLinkCode
       .post(baseUrl)
       .withHeaders(X_XSRF_TOKEN.key, X_XSRF_TOKEN.value)
+      .withCookies(X_XSRF_TOKEN.key, X_XSRF_TOKEN.value)
       .withJson({
-        requestTime: requestTime,
+        requestTime: null,
         request: {
-          transactionId: 'reused_completed_transaction_id',
+          transactionId: transactionId,
         },
       })
 );
+
+// Scenario: Not able to generate link code because of invalid xsrf token
+// Given and others Then for this scenario are written in the aforementioned example
+When(
+  'Send POST \\/linked-authorization\\/link-code request with given invalid X-XSRF-TOKEN header, transactionId and requestTime',
+  () =>
+    specWalletGenerateLinkCode
+      .post(baseUrl)
+      .withHeaders(X_XSRF_TOKEN.key, 'invalid value')
+      .withCookies(X_XSRF_TOKEN.key, 'invalid value')
```

Comment:
In this case the header and cookie should have different value to be consider "invalid X-XSRF-TOKEN header"